src/_guides/language/effective-dart/usage.md

```diff
@@ -376,8 +376,9 @@ numbers.removeAt(1); // Now it only contains integers.
 var ints = List<int>.from(numbers);
 {% endprettify %}
 
-But if your goal is just to copy the iterable and preserve its original type, or
-you don't care about the type, then use `toList()`.
+If your goal is to copy the iterable and preserve its runtime element type, or
+you don't care about the type, then use `toList()`. If your goal is to copy the
+iterable and preserve its static element type, use `List.of()`.
```

Comment:
This is better, but now that I look at the whole guideline, the "two obvious ways" part is clearly out of date too since now we also have `List.of()`. This really needs more work.

Reply:
Also "preserve its static element type" doesn't mean what I think it's intended to say. I read it as making another list with the same *static* type (which both `.toList()` and `List.of` will do). To say what I think this intends to say, I'd use "... and reify its static element type, ...".
Most likely users won't understand that.

On top of that, what you reify is not always the *static type*, but rather the *static context type* if there is one. If you do:
```dart
var x = Iterable<int>.generate(...);
List<num> l = List.of(x);
```
then `List.of` will infer `num` as its type argument. 
The case described here is closer to:
```dart
Iterable<num> x = Iterable<int>.generate(...);
var l = List.of(x);
```
In this case, with no context type, `List.of` infers its type argument from the static type of its argument.

`List.of` can only convert an iterable of T to a list of a super-type of T. The only reason to do that, rather than `toList`, is because you want to *add* elements to (or modify elements of) the created list. So, the text should mention that.

Maybe:

> If you need to create a list that will be modified (added or written to), you need it to have a specific element type, not just any subtype of that. For that, you can usually use the `List.of` constructor, which will infer the runtime element type from the context or the static type its argument.